lib/compiler-singlepass/src/codegen_x64.rs

```diff
@@ -325,6 +327,28 @@ impl<'a> FuncGen<'a> {
         self.mark_instruction_address_end(offset);
     }
 
+    fn emit_trap(&mut self, code: TrapCode) {
```

Comment:
Maybe a bit fancier would be if the trap PC would reflect the actual place in JITed code, not catch-all labels. Fix is pretty simple, we just need to init RDI to trap location before jump, but not sure if it's too important.

Reply:
Alternative option is to inline trap_handler to the call site, not jump, but that would mean larger compiled binary, so not desirable.